### ISSUE TITLE
Provide option to add "Open in Windows Terminal (Admin)" to Explorer context menu

### ISSUE THREAD

**Issue by TheAndyMac** [microsoft/terminal#9903]
# Description of the new feature/enhancement

As well as having the option to Open in Windows Terminal as a right-click in Explorer (which should be configurable as discussed in #9902 and #6113 ) it would be really helpful to have a configurable option to ALSO have an "Open in Windows Terminal (Admin)" option to quickly open up Windows Terminal with Admin rights to be able to access those commands which need it.

# Proposed technical implementation details (optional)

Add an option in the configuration to enable/disable an "Open in Windows Terminal (Admin)" context menu option.

**Comment by zadjii-msft** [comment]
Yea, we'll make sure to include this in the configuration options. This is technically different than #6111 so I'll leave it open

**Comment by Dhyfer1** [comment]
It has been 19 months since the creation of this issue and I also need Windows Terminal to be able to run as admin from the context menu. Why is there still no progress on this? or this isn't a priority for the developers behind Windows Terminal?

**Comment by zadjii-msft** [comment]
Indeed, this isn't an immediate priority for us. You'll note that across the repo, there are plenty of older issues, with more upvotes, which are currently a higher priority for us. 

If you'd like to help, we'd be happy to help point you in the right direction! This is unfortunately kinda a tricky problem, because we need to solve two problems:
* We need to add support for the shell context menu code to load our settings model. I don't know if we can do that easily currently. 
* We need to find a way to add settings to configure what our context menu settings should be. What shape do we want these settings to be? `contextMenu.showElevatedEntry` & `contextMenu.showProfilePicker`? `contextMenu.showElevatedProfilePicker`? We need some sort of design for that.  
* Putting nested entries in the menu is non-trivial, https://github.com/microsoft/terminal/issues/6111#issuecomment-1252946834 has some notes. At least I'm not enough of a shell32 guru to know. 

Figuring out the first bullet point is probably the hardest part. If you can get our settings loaded in the context menu code, then it's easy[^1] to add settings to control this. 

[^1]: Of course, naming in software is a **H**ard problem, but adding settings is easy once you've figured that out 😋

**Comment by Mujtaba0150** [comment]
An easy workaround for someone willing to mess with the registry would be to add a registry key to the right click context menu to use PowerShell to open wt as an admin using the UAC prompt using this command:
Start-Process wt.exe -Verb RunAs

**Comment by akopetsch** [comment]
I've created a workaround that adds an 'Open in Terminal as administrator' option to the extended context menu in Windows Explorer. You can find it here: [akopetsch/WindowsTerminalAdmin](https://github.com/akopetsch/WindowsTerminalAdmin).

**Comment by MarvinKlein1508** [comment]
Is there also any way to add it to the new context menu in Windows 11?

**Comment by eiqnepm** [comment]
Windows Terminal is already in the Windows 11 context menu

Are you sure you haven't disabled it by accident? 

https://www.tenforums.com/tutorials/176295-add-remove-open-windows-terminal-context-menu-windows-10-a.html

**Comment by MarvinKlein1508** [comment]
I'Ve applied the registry patch and restarted my PC. Still no context menu:

![Image](https://github.com/user-attachments/assets/0d3219f7-1e40-46bd-9e33-99c642765f46)
